app/plugin_api.go

```diff
@@ -852,3 +852,33 @@ func (api *PluginAPI) PluginHTTP(request *http.Request) *http.Response {
 	api.app.ServeInterPluginRequest(responseTransfer, request, api.id, destinationPluginId)
 	return responseTransfer.GenerateResponse()
 }
+
+func (api *PluginAPI) SendPluginEvent(event string, payload interface{}) *model.AppError {
+	pluginEnvironment := api.app.GetPluginsEnvironment()
+	subscriptionsForEvent := pluginEnvironment.GetPluginSubscriptionsForEvent(event)
+	errorList := []string{}
+
+	for _, subscribedPlugin := range subscriptionsForEvent {
+		error := api.app.servePluginEvent(event, payload, subscribedPlugin, api.manifest.Id)
+
+		if error != nil {
+			errorList = append(errorList, subscribedPlugin)
```

Comment:
In addition to returning the aggregate error, I think the errors should be logged individually here, as the error messages may be different per plugin.